src/Playwright.Tests/BrowserTypeConnectTests.cs

```diff
@@ -80,71 +82,396 @@ public void TearDown()
-        [Ignore("SKIP WIRE")]
-        public void ShouldRespectSelectors()
+        [PlaywrightTest("browsertype-connect.spec.ts", "should error when saving download after deletion")]
+        public async Task ShouldErrorWhenSavingDownloadAfterDeletion()
         {
+            Server.SetRoute("/download", context =>
+            {
+                context.Response.Headers["Content-Type"] = "application/octet-stream";
+                context.Response.Headers["Content-Disposition"] = "attachment; filename=file.txt";
+                return context.Response.WriteAsync("Hello world");
+            });
+
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var page = await browser.NewPageAsync(new() { AcceptDownloads = true });
+            await page.SetContentAsync($"<a href=\"{Server.Prefix}/download\">download</a>");
+            var downloadTask = page.WaitForDownloadAsync();
+
+            await TaskUtils.WhenAll(
+                downloadTask,
+                page.ClickAsync("a"));
+
+            using var tmpDir = new TempDirectory();
+            string userPath = Path.Combine(tmpDir.Path, "download.txt");
+            var download = downloadTask.Result;
+            await download.DeleteAsync();
+            var exception = await PlaywrightAssert.ThrowsAsync<PlaywrightException>(() => download.SaveAsAsync(userPath));
+            StringAssert.Contains("Target page, context or browser has been closed", exception.Message);
+        }
+
+        [PlaywrightTest("browsertype-connect.spec.ts", "should be able to connect when the wsEndpont is passed as the first argument")]
+        public async Task ShouldConnectWhenWsEndpointIsPassedAsFirstArgument()
+        {
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var context = await browser.NewContextAsync();
+            var page = await context.NewPageAsync();
+            var result = await page.EvaluateAsync("1 + 2");
+            Assert.AreEqual(result.ToString(), "3");
+            await browser.CloseAsync();
         }
 
+        [PlaywrightTest("browsertype-connect.spec.ts", "should save har")]
+        public async Task ShouldSaveHar()
+        {
+            using var tempDirectory = new TempDirectory();
+            var harPath = tempDirectory.Path + "/test.har";
+            var browser = await BrowserType.ConnectAsync(_browserServer.WSEndpoint);
+            var context = await browser.NewContextAsync(new()
+            {
+                RecordHarPath = harPath
+            });
+
+            var page = await context.NewPageAsync();
+            await page.GotoAsync(Server.EmptyPage);
+            await context.CloseAsync();
+            await browser.CloseAsync();
+
+
+            Assert.That(harPath, Does.Exist);
+            var logString = System.IO.File.ReadAllText(harPath);
+            StringAssert.Contains(Server.EmptyPage, logString);
+        }
 
```

Comment:
"should record trace with sources" test is missing

Reply:
Added this test.